src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -312,29 +321,52 @@
-    </ItemGroup>
+    <PropertyGroup>
+      <VersionPadding/>
+      <VersionComparisonDate/>
+    </PropertyGroup>
       
-    <!-- Windows Exec command requires DOS escaping for the percent sign since it secretly runs the process in a shell instead of calling createprocess. -->
     <PropertyGroup>
-      <GitCommand Condition="'$(OsEnvironment)'=='Windows_NT'">git show -s --format=^%25%25cd --date=short HEAD</GitCommand>
-      <GitCommand Condition="'$(OsEnvironment)'!='Windows_NT'">git show -s --format=%25cd --date=short HEAD</GitCommand>
+      <VersionFileContent>
+        <![CDATA[<?xml version="1.0" encoding="utf-8"?>
+<!-- This is a generated file. $(VersionSourceComment) Seed Date is $(VersionSeedDate). -->
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VersionPropsImported>true</VersionPropsImported>
+    <BuildNumberMajor Condition="'%24(BuildNumberMajor)' == ''">$(BuildNumberMajor)</BuildNumberMajor>
+    <BuildNumberMinor Condition="'%24(BuildNumberMinor)' == ''">$(BuildNumberMinor)</BuildNumberMinor>
+    <LatestCommit Condition="'%24(LatestCommit)' == ''">$(LatestCommit)</LatestCommit>
+    <BuiltByString Condition="'%24(BuiltByString)' == ''">$(BuiltByString)</BuiltByString>
+  </PropertyGroup>
+</Project>
+]]>
+      </VersionFileContent>
     </PropertyGroup>
+    
+    <!-- Since by default the file will get dropped at the obj dir, make sure that the dir is created already or else WriteLinesToFile will error. -->
+    <MakeDir Condition="!Exists('$(BuildVersionFilePath)')" Directories="$(BuildVersionFilePath)" />
       
-    <Exec Command="$(GitCommand)" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
-      <Output TaskParameter="ConsoleOutput" PropertyName="VersionSeedDate" />
-      <Output TaskParameter="ExitCode" PropertyName="GitCommandExitCode" />
-    </Exec>
-    <PropertyGroup Condition="'$(GitCommandExitCode)'!='0'">
-      <VersionSeedDate>%(VersionTargetsFile.ModifiedTime)</VersionSeedDate>
-      <VersionSeedSourceComment>VersionSeedDate was produced by getting the timestamp of versioning.targets.</VersionSeedSourceComment>
+    <WriteLinesToFile
+      ContinueOnError="WarnAndContinue"
+      Condition="!Exists('$(BuildVersionFile)')"
+      File="$(BuildVersionFile)"
+      Lines="$(VersionFileContent)"
+      Overwrite="true" />
+    
+    <!-- Delete old BuildVersion.props files -->
+    <ItemGroup>
+      <OldBuildVersionFiles Include="$(BuildVersionFilePath)BuildVersions-*.props" Exclude="$(BuildVersionFile)" />
+    </ItemGroup>
+    <Delete Files="@(OldBuildVersionFiles)" TreatErrorsAsWarnings="true"/>
+    <ItemGroup>
+      <OldBuildVersionFiles Remove="@(OldBuildVersionFiles)"/>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <AssemblyFileVersion>$(MajorVersion).$(MinorVersion).$(BuildNumberMajor).$(BuildNumberMinor)</AssemblyFileVersion>
```

Comment:
Can you add a comment here that says we need to redefine AssemblyFileVersion because we just calculated it?
Have you looked for any other derived properties that need to be reset?